### ISSUE TITLE
Don’t recommend omitting the block argument to define_method

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#3499]
When using `define_method` to define a method, the block arguments given will turn into method arguments on the defined method. So when the `Lint/UnusedBlockArgument` cop recommends that I “can omit the argument”, it is actually suggesting that I change the arity of a method without changing its callers.

Running RuboCop with the `-a` flag does the right thing and prefixes the unused argument with an underscore.

--------

## Expected behavior

Recommend that I use `_` prefix for unused block argument to `define_method`.

## Actual behavior

Recommend that I use delete the “unused” block argument to `define_method`.

## Steps to reproduce the problem

Given a file test.rb:
```rb
# encoding: utf-8
# frozen_string_literal: true

define_method('foo') do |bar|
  puts 'Completely ignores the argument'
end

foo(42)
```

Running RuboCop gives this output:
```
Inspecting 1 file
W

Offenses:

test.rb:4:26: W: Unused block argument - bar. You can omit the argument if you don't care about it.
define_method('foo') do |bar|
                         ^^^

1 file inspected, 1 offense detected
```

## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-darwin15)
```
